### DISCUSSION TITLE
Example of decrypting a JWT?

### DISCUSSION THREAD

**Discussion by anderspitman** [lestrrat-go/jwx#1003] [General]
The docs have a nice example of how to encrypt and sign a JWT. Is there an example anywhere of how to decrypt/parse a JWT that has been encrypted this way?

**Comment by lestrrat** [comment]
Use `jwe.Decrypt`, and then `jwt.Parse`. Detecting signed JWTs is baked into `jwt.Parse` because it's almost ubiquitous, but JWE isn't because 1) it's not as common as signed JWTs, and 2) if `jwt.Parse` allowed detecting signing and encryption, then it would also have to detect _nested_ signed/encrypted messages, which makes the process much harder.

Examples for decryption can be found in the [docs](https://github.com/lestrrat-go/jwx/blob/develop/v2/docs/03-jwe.md) and [examples directory](https://github.com/lestrrat-go/jwx/blob/develop/v2/examples)

**Reply by anderspitman** [reply]
Working great, thanks!
